Enable Objectice-C ARC for JavaScriptCore

SiliconA-Z opened the pull request:
#### cfb8715dd854d4220ae9f5b554915d5ebf8eb286
<pre>
Enable Objectice-C ARC for JavaScriptCore
<a href="https://bugs.webkit.org/show_bug.cgi?id=251797">https://bugs.webkit.org/show_bug.cgi?id=251797</a>

Reviewed by NOBODY (OOPS!).

This patch finally enables JavaScriptCore to use Automated Reference
Counting, which means we do not have to rely on smart pointers for
Objective-C objects.

Unfortunately, there were 3 files that could not be converted to ARC.

List of files that were not migrated and the reasoning for each:

1. JSAPIWrapperObject.mm: ARC is unaware of what happens in releaseSoon,
   and as a result, it inserts release code, overreleasing the object.

2. ObjC bridge uses NSInvocation, which means ARC is a non-starter
   for this file in addition to conditional memory management regarding
   the unwrapping of Obj-C objects. It is also unaware of releaseSoon,
   and as a result, it inserts release code, overreleasing the object.

3. JSContext.mm manually retains and releases objects via callbacks,
   which ARC cannot account for.

* Source/JavaScriptCore/API/JSAPIWrapperObject.h:
(JSC::JSAPIWrapperObject::wrappedObject):
* Source/JavaScriptCore/API/JSAPIWrapperObject.mm:
(JSAPIWrapperObjectHandleOwner::finalize):
(JSAPIWrapperObjectHandleOwner::isReachableFromOpaqueRoots):
(JSC::JSAPIWrapperObject::setWrappedObject):
(JSC::JSAPIWrapperObject::visitChildrenImpl):
* Source/JavaScriptCore/API/JSContext.mm:
(-[JSContext ensureWrapperMap]):
(-[JSContext init]):
* Source/JavaScriptCore/API/JSManagedValue.h:
* Source/JavaScriptCore/API/JSManagedValue.mm:
(-[JSManagedValue init]):
(-[JSManagedValue dealloc]):
* Source/JavaScriptCore/API/JSScript.h:
* Source/JavaScriptCore/API/JSScriptInternal.h:
* Source/JavaScriptCore/API/JSValue.mm:
(-[JSValue dealloc]):
(-[JSValue initWithValue:inContext:]):
(-[JSValue init]): Deleted.
* Source/JavaScriptCore/API/JSVirtualMachine.mm:
(-[JSVirtualMachine dealloc]):
(scanExternalObjectGraph):
(scanExternalRememberedSet):
* Source/JavaScriptCore/API/JSVirtualMachineInternal.h:
* Source/JavaScriptCore/API/JSWrapperMap.h:
* Source/JavaScriptCore/API/JSWrapperMap.mm:
(makeWrapper):
(copyPrototypeProperties):
(-[JSObjCClassInfo dealloc]):
(-[JSObjCClassInfo allocateConstructorAndPrototypeInContext:]):
(-[JSObjCClassInfo wrapperForObject:inContext:]):
(-[JSWrapperMap initWithGlobalContextRef:]):
(-[JSWrapperMap objcWrapperForJSValueRef:inContext:]):
(tryUnwrapObjcObject):
* Source/JavaScriptCore/API/ObjCCallbackFunction.mm:
(objCCallbackFunctionForMethod):
* Source/JavaScriptCore/API/tests/CurrentThisInsideBlockGetterTest.mm:
(ConstructorFinalize):
* Source/JavaScriptCore/Configurations/Base.xcconfig:
* Source/JavaScriptCore/JavaScriptCore.xcodeproj/project.pbxproj:
* Source/JavaScriptCore/SourcesCocoa.txt:
* Source/WTF/wtf/RetainPtr.h:
(WTF::RetainPtr::RetainPtr):
(WTF::RetainPtr::checkType): Deleted.
</pre><!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/98d0c12ac2fd03a85d79bf31baa0ced8ce3f92b1

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [  ~~🧪 style~~](https://ews-build.webkit.org/#/builders/38/builds/65678 "EWS skipped this build as PR had skip-ews label when EWS attempted to process it.") | [  ~~🛠 ios~~](https://ews-build.webkit.org/#/builders/48/builds/45051 "EWS skipped this build as PR had skip-ews label when EWS attempted to process it.") | [  ~~🛠 mac~~](https://ews-build.webkit.org/#/builders/55/builds/18297 "EWS skipped this build as PR had skip-ews label when EWS attempted to process it.") | [  ~~🛠 wpe~~](https://ews-build.webkit.org/#/builders/5/builds/69704 "EWS skipped this build as PR had skip-ews label when EWS attempted to process it.") | [  ~~🛠 wincairo~~](https://ews-build.webkit.org/#/builders/59/builds/16287 "EWS skipped this build as PR had skip-ews label when EWS attempted to process it.") 
| | [  ~~🛠 ios-sim~~](https://ews-build.webkit.org/#/builders/49/builds/52850 "EWS skipped this build as PR had skip-ews label when EWS attempted to process it.") | [  ~~🛠 mac-AS-debug~~](https://ews-build.webkit.org/#/builders/61/builds/16569 "EWS skipped this build as PR had skip-ews label when EWS attempted to process it.") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/5/builds/69704 "EWS skipped this build as PR had skip-ews label when EWS attempted to process it.") | [  ~~🧪 wincairo-tests~~](https://ews-build.webkit.org/#/builders/59/builds/16287 "EWS skipped this build as PR had skip-ews label when EWS attempted to process it.") 
| [  ~~🧪 webkitperl~~](https://ews-build.webkit.org/#/builders/11/builds/68745 "EWS skipped this build as PR had skip-ews label when EWS attempted to process it.") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/49/builds/52850 "EWS skipped this build as PR had skip-ews label when EWS attempted to process it.") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/55/builds/18297 "EWS skipped this build as PR had skip-ews label when EWS attempted to process it.") | [  ~~🧪 api-wpe~~](https://ews-build.webkit.org/#/builders/5/builds/69704 "EWS skipped this build as PR had skip-ews label when EWS attempted to process it.") | 
| | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/49/builds/52850 "EWS skipped this build as PR had skip-ews label when EWS attempted to process it.") | [  ~~🧪 mac-wk1~~](https://ews-build.webkit.org/#/builders/55/builds/18297 "EWS skipped this build as PR had skip-ews label when EWS attempted to process it.") | [  ~~🛠 wpe-cairo~~](https://ews-build.webkit.org/#/builders/65/builds/15163 "EWS skipped this build as PR had skip-ews label when EWS attempted to process it.") | 
| [  ~~🛠 🧪 jsc~~](https://ews-build.webkit.org/#/builders/20/builds/58792 "EWS skipped this build as PR had skip-ews label when EWS attempted to process it.") | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/49/builds/52850 "EWS skipped this build as PR had skip-ews label when EWS attempted to process it.") | [  ~~🧪 mac-wk2~~](https://ews-build.webkit.org/#/builders/55/builds/18297 "EWS skipped this build as PR had skip-ews label when EWS attempted to process it.") | [  ~~🛠 gtk~~](https://ews-build.webkit.org/#/builders/2/builds/71410 "EWS skipped this build as PR had skip-ews label when EWS attempted to process it.") | 
| [  ~~🛠 🧪 jsc-arm64~~](https://ews-build.webkit.org/#/builders/12/builds/64922 "EWS skipped this build as PR had skip-ews label when EWS attempted to process it.") | [  ~~🛠 vision~~](https://ews-build.webkit.org/#/builders/87/builds/9633 "EWS skipped this build as PR had skip-ews label when EWS attempted to process it.") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/61/builds/16569 "EWS skipped this build as PR had skip-ews label when EWS attempted to process it.") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/2/builds/71410 "EWS skipped this build as PR had skip-ews label when EWS attempted to process it.") | 
| | [  ~~🛠 vision-sim~~](https://ews-build.webkit.org/#/builders/86/builds/9665 "EWS skipped this build as PR had skip-ews label when EWS attempted to process it.") | [  ~~🧪 mac-wk2-stress~~](https://ews-build.webkit.org/#/builders/55/builds/18297 "EWS skipped this build as PR had skip-ews label when EWS attempted to process it.") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/2/builds/71410 "EWS skipped this build as PR had skip-ews label when EWS attempted to process it.") | 
| | [⏳ 🧪 vision-wk2 ](https://ews-build.webkit.org/#/builders/visionOS-1-Simulator-WK2-Tests-EWS "Waiting in queue, processing has not started yet") | [  ~~🧪 mac-intel-wk2~~](https://ews-build.webkit.org/#/builders/55/builds/18297 "EWS skipped this build as PR had skip-ews label when EWS attempted to process it.") | [  ~~🛠 jsc-armv7~~](https://ews-build.webkit.org/#/builders/35/builds/86689 "EWS skipped this build as PR had skip-ews label when EWS attempted to process it.") | 
| | [  ~~🛠 tv~~](https://ews-build.webkit.org/#/builders/44/builds/40859 "EWS skipped this build as PR had skip-ews label when EWS attempted to process it.") | | [  ~~🧪 jsc-armv7-tests~~](https://ews-build.webkit.org/#/builders/35/builds/86689 "EWS skipped this build as PR had skip-ews label when EWS attempted to process it.") | 
| | [  ~~🛠 tv-sim~~](https://ews-build.webkit.org/#/builders/46/builds/41935 "EWS skipped this build as PR had skip-ews label when EWS attempted to process it.") | | | 
| | [  ~~🛠 watch~~](https://ews-build.webkit.org/#/builders/43/builds/43118 "EWS skipped this build as PR had skip-ews label when EWS attempted to process it.") | | | 
| | [  ~~🛠 watch-sim~~](https://ews-build.webkit.org/#/builders/45/builds/41679 "EWS skipped this build as PR had skip-ews label when EWS attempted to process it.") | | | 
<!--EWS-Status-Bubble-End-->

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/b2e35e73d6ba0bc5254eeaa77d187a3c2291fd6f. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/13105

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/0bb86fc654500f83f71c4822753bd6b4717a4198. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/13105

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/0bef0ee144e23309f77abc7fb6490c9cdd9d18eb. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/13105

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/214f5ea07060a2d5bd19be651e60212dead95e97. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/13105

AtariDreams commented:
I think the Xcode compiler is bugged. It keeps omitting errors saying there is implicit casting but I see none of it

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/af08da8a9494aa0f34ceb31d835ccc810524d719. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/13105

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/27f57770ad4f3efc6be2c259b0033edaffc6d326. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/13105

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/b966719609eb526d848c39e4025c023ae663f2d6. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/13105

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/9262d5482cfd7b1535cffe53c0c7431b5424465b. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/13105

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/fdca48a3b8c54917eec03453ef6824e80121cde1. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/13105

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/e7ae7ce8909f6db38b2b40fa974ec8767813b80b. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/13105

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/9ee0043063eace455fca4e4e54f08269db20c854. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/13105

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/9bc2d73d1e5a4c45e2bd1537359d6361cd82b25f. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/13105

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/113cc79418ceebc30c710baa9c9e5ccc17619433. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/13105

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/497b9efa451c1e768e57c790cdadd55688589b43. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/13105

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/751cb7cde080b043255546eb80d29d59ff727638. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/13105

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/94c2b427019f637d53a2563a09f2f19ee857fd7e. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/13105

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/5d90df52a47a0934411af83f7718829a2aac610a. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/13105

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/e50a2e48e1b38f40cdf3e2b74dcd5712bd923cd6. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/13105

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/0ae86a16501fccafa469cec2e4a0e881679507a1. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/13105

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/b745f50c660acb0155cacfc2ae847152c8ea5a11. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/13105

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/7e354cc4472aacfa32dd2231f70064f8f84bdfa8. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/13105

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/8028f6a0fbbeff3735d698364a3d32e94bb7d1a3. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/13105

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/c1ee25f35ca7e4382d7a51f0d141a343d541194c. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/13105

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/52c349115789231855d0a2b32076522255f0e0dc. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/13105

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/54ba0ecb933c846f686199334a0f3858d8ced5b9. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/13105

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/e408a3a636d47c25df2d0fdca90c939551c11543. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/13105

LumioseSil committed:
Enable Objectice-C ARC for JavaScriptCore https://bugs.webkit.org/show_bug.cgi?id=251797

Reviewed by NOBODY (OOPS!).

This patch finally enables JavaScriptCore to use Automated Reference
Counting, which means we do not have to rely on smart pointers for
Objective-C objects.

Unfortunately, there were 3 files that could not be converted to ARC.

List of files that were not migrated and the reasoning for each:

1. JSAPIWrapperObject.mm: ARC is unaware of what happens in releaseSoon,
   and as a result, it inserts release code, overreleasing the object.

2. ObjC bridge uses NSInvocation, which means ARC is a non-starter
   for this file in addition to conditional memory management regarding
   the unwrapping of Obj-C objects. It is also unaware of releaseSoon,
   and as a result, it inserts release code, overreleasing the object.

3. JSContext.mm manually retains and releases objects via callbacks,
   which ARC cannot account for.

* Source/JavaScriptCore/API/JSAPIWrapperObject.h:
(JSC::JSAPIWrapperObject::wrappedObject):
* Source/JavaScriptCore/API/JSAPIWrapperObject.mm:
(JSAPIWrapperObjectHandleOwner::finalize):
(JSAPIWrapperObjectHandleOwner::isReachableFromOpaqueRoots):
(JSC::JSAPIWrapperObject::setWrappedObject):
(JSC::JSAPIWrapperObject::visitChildrenImpl):
* Source/JavaScriptCore/API/JSContext.mm:
(-[JSContext ensureWrapperMap]):
(-[JSContext init]):
* Source/JavaScriptCore/API/JSManagedValue.h:
* Source/JavaScriptCore/API/JSManagedValue.mm:
(-[JSManagedValue init]):
(-[JSManagedValue dealloc]):
* Source/JavaScriptCore/API/JSScript.h:
* Source/JavaScriptCore/API/JSScriptInternal.h:
* Source/JavaScriptCore/API/JSValue.mm:
(-[JSValue dealloc]):
(-[JSValue initWithValue:inContext:]):
(-[JSValue init]): Deleted.
* Source/JavaScriptCore/API/JSVirtualMachine.mm:
(-[JSVirtualMachine dealloc]):
(scanExternalObjectGraph):
(scanExternalRememberedSet):
* Source/JavaScriptCore/API/JSVirtualMachineInternal.h:
* Source/JavaScriptCore/API/JSWrapperMap.h:
* Source/JavaScriptCore/API/JSWrapperMap.mm:
(makeWrapper):
(copyPrototypeProperties):
(-[JSObjCClassInfo dealloc]):
(-[JSObjCClassInfo allocateConstructorAndPrototypeInContext:]):
(-[JSObjCClassInfo wrapperForObject:inContext:]):
(-[JSWrapperMap initWithGlobalContextRef:]):
(-[JSWrapperMap objcWrapperForJSValueRef:inContext:]):
(tryUnwrapObjcObject):
* Source/JavaScriptCore/API/ObjCCallbackFunction.mm:
(objCCallbackFunctionForMethod):
* Source/JavaScriptCore/API/tests/CurrentThisInsideBlockGetterTest.mm:
(ConstructorFinalize):
* Source/JavaScriptCore/Configurations/Base.xcconfig:
* Source/JavaScriptCore/JavaScriptCore.xcodeproj/project.pbxproj:
* Source/JavaScriptCore/SourcesCocoa.txt:
* Source/WTF/wtf/RetainPtr.h:
(WTF::RetainPtr::RetainPtr):
(WTF::RetainPtr::checkType): Deleted.

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/cfb8715dd854d4220ae9f5b554915d5ebf8eb286. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/13105

hortont424 commented on Source/JavaScriptCore/JavaScriptCore.xcodeproj/project.pbxproj:
```diff
@@ -12658,13 +12661,15 @@
 			isa = XCBuildConfiguration;
 			baseConfigurationReference = 1C9051430BA9E8A70081E9D0 /* JavaScriptCore.xcconfig */;
 			buildSettings = {
+				CLANG_ENABLE_OBJC_ARC = YES;
```
This should definitely not be here, but in some xcconfig file

AtariDreams replied:
Fixed!

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/3a137769f967c40cc1f986d48e9b350b842251e4. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/13105

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/de0dc313b95d99691901e3516c655aa672988649. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/13105

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/d83e53857418cbf37261d4b3d281956f5b658579. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/13105

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/ca436a54eebde30b73c403262c61147447cd10e4. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/13105

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/4c84bfb7c5e6822b879b1f4e2ebc683f0459988d. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/13105

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/a3b9ca2e9d2b953e9e0c4f91625d055f1b265a79. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/13105

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/bbc79a62361844084454d3f3417725f14c422df6. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/13105

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/d1f1080a07b37d074e1b45f2f1fb3dac67015bd6. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/13105

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/7353e6c712876660e0ad4adf2ce7fcdc45167c3e. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/13105

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/e8bd3cd8a76e7347c52e099ad1abff4a0b56532f. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/13105

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/d0006f62e0ede872c8df418f18352d2591eff0e8. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/13105

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/8ed1b924a1693125df17fbb0d6d46cb71e8f48a9. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/13105

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/1acbaf16b4a9e42b1898afdc542f51017649a348. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/13105

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/942e3bf74ccd57d748ce1f6d9665bd7bd6ceb360. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/13105

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/49e321c6280ef471fc9697af34b233c4a4778e5b. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/13105

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/3496d886e7bbfbfef04a0e3680c99137a68b1829. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/13105

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/77833c1c78867edfef7fbd109e753add7f4292b6. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/13105

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/b2ebcd81ee8a9b938ee22c2a08fc4d5ea4053ad4. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/13105

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/ffa09fdef114c580095f8c4c17d71ffe89b406ea. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/13105

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/5488bfc94cf314accce97b21ec7b354ef77a2004. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/13105

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/d01e7482511aea7330c28bc14f6d2f6bc189fdb6. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/13105

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/2b9b6ccb47f6b96cfb77ea51dbf9a8ae979a38a8. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/13105

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/468b739ee806d9018306f9b6a33c08721bca4638. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/13105

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/d11684bde227c5f1e5cfae045fcdb327d742d64b. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/13105

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/700d38b9364b978a27f3ca122144a61b0adb92ee. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/13105

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/2eff3aac13e3025dd29ffb8baef4c49fece82f68. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/13105

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/9f1b8bfa0ddd6723e524f3661f2fc32a4f3d7134. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/13105

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/dc1b6478e1770f15b6e8ee35435d211152ef122b. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/13105

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/9f1842983e7b4f0db3ee3ac1b7fa6782b863ab8a. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/13105

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/0b55bd53c25348c2658b33d83793f5e5d6418975. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/13105

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/7c883e70d8a3399d6a991bf94420715c35d87708. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/13105

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/d5ba0c7213cc191f0754da7462285c7451022a3c. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/13105

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/a3902372fe691d9fe05a592d9b055730db9c134f. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/13105

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/2ce06a856327c1ce1f3d1b8dab4aacadc5fa0433. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/13105

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/6af91c66bc46265309cb41b0f5298440474d9608. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/13105

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/04725a9a0c03bfc7644d47e66637e7daad2b8d77. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/13105

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/b331b02ab19f0b5649912cb08df674702b0af781. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/13105

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/c8f87310dd238e2e0bc67510a4b4064ecb7a1523. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/13105

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/12ddac4dc1ee98981598f54eda252a00f1e2a18d. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/13105

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/29e70fe8383a7a63a263567795c219beaeac094d. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/13105

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/e55301c104fb0f9ac35f5d0f0f39c5b877d83660. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/13105

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/3897a941dc6de914aa9064bfd70eb5d9e4620286. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/13105

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/d676e028be782a0dae93ce9a3ae7cdd30d2d7552. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/13105

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/d3533e76aa53e5ec720ba059f03d50b57d38fffb. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/13105

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/3dbf2cf3df2c6ed3efd972f9eb1bb2e8e7e346f9. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/13105

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/255f02df04fa6b38f36a175a5ee9e430f7b6cd85. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/13105

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/fa65890e36e8dcbc029f081bd0948ef380abc0ff. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/13105

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/7590fb7ac156cea20fafa9449115403ce86ff71c. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/13105

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/3e7febbc104bf5448be3598b1b8d127b6759d80c. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/13105

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/9964ecc359155566912baa296de6cb7b504e9b13. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/13105

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/923b06a918f2f47e5ecdc59a5d2ea0f0b015a4a2. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/13105

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/22a1f669995f9b080877b436b7d488562be2dfab. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/13105

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/ea57e5d7b2e9ddf9665f84a8a03bb42e2649bdfe. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/13105

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/e0a5f168f617c525fe197ce39481b4f34e5cda72. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/13105

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/19873233962940c21c872ea46a9c5f625ff27b44. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/13105

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/9bfddc5f971fadcdbac81cc4934b33a90dd0a9c4. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/13105

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/2a836bd16480494d3d5b0b7e306f77e252841e13. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/13105

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/99290ea3ca6dc9d20c6a3cc5bdd827c4bc50c271. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/13105

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/4382541167dbff4748df127100b0de2bf3c395ed. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/13105

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/9b8bf2d5e51555887d8984f9cf6457309fb32221. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/13105

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/cd5e2dc6f25427b1dd899eeb90a1ad420783eeeb. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/13105

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/45b636a2ceae8a113bfc303769d1e27174366462. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/13105

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/98d0c12ac2fd03a85d79bf31baa0ced8ce3f92b1. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/13105

RSilicon commented:
Way smaller in scope and more refined
